src/components/avatar/avatar.tsx

```diff
@@ -25,21 +29,39 @@ export const Avatar: FC<AvatarProps> = ({
   src,
   alt,
   fallbackText,
+  editable = false,
+  onEditClick,
 }) => (
-  <AvatarPrimitive.Root
-    className={twMerge(
-      "relative flex shrink-0 overflow-hidden rounded-full",
-      sizeClassMap[size],
-      border && "border-[6px] border-neutral-100",
+  <div className="relative inline-block">
+    <AvatarPrimitive.Root
+      data-testid="avatar-root"
+      className={twMerge(
+        "relative flex shrink-0 overflow-hidden rounded-full",
+        sizeClassMap[size],
+        border && "border-[6px] border-neutral-100",
+      )}
+    >
+      <AvatarPrimitive.Image
+        src={src}
+        alt={alt}
+        className="aspect-square size-full object-cover"
+      />
+      <AvatarPrimitive.Fallback className="bg-primary-100 text-primary flex size-full items-center justify-center rounded-full font-medium">
+        {fallbackText}
+      </AvatarPrimitive.Fallback>
+    </AvatarPrimitive.Root>
+
+    {editable && (
+      <div className="absolute bottom-0 right-0">
+        <Button
+          variant="neutral"
+          size="default"
+          isCircular
+          onClick={onEditClick}
```

Comment:
We have to pass either a button label or the aria attribute to remain accessible (However, the "active" state of the button is hard to see anyway, but that's by design)

```suggestion
          onClick={onEditClick}
          aria-label="Edit Avatar"
```